backend/pkg/api/connect/errors/common_errors.go

```diff
@@ -27,3 +27,14 @@ func NewRedpandaAdminAPINotConfiguredError() *connect.Error {
 		NewHelp(NewHelpLinkConsoleReferenceConfig()),
 	)
 }
+
+// NewRedpandaFeatureNotSupportedError is a standard error to return if an endpoint
+// requires the Redpanda feature that is not supported with current running version and instance.
+func NewRedpandaFeatureNotSupportedError(feature string) *connect.Error {
+	return NewConnectError(
+		connect.CodeUnimplemented,
+		errors.New("redpanda version does not support feature "+feature),
+		NewErrorInfo(v1alpha1.Reason_REASON_FEATURE_NOT_SUPPORTED.String()),
+		NewHelp(NewHelpLinkConsoleReferenceConfig()),
+	)
+}
```

Comment:
Maybe we can use a more generic error here instead? We already have something similar for Kafka API endpoints if the target cluster version does not implement a certain Kafka API version.

Reply:
I am not sure I understand exactly what you mean by a more generic error? Can you elaborate?